addons/l10n_in_sale/models/sale_order.py

```diff
@@ -20,6 +20,49 @@ class SaleOrder(models.Model):
             ('uin_holders', 'UIN Holders'),
         ], string="GST Treatment", readonly=False, compute="_compute_l10n_in_gst_treatment", store=True)
 
+    @api.depends('partner_id', 'partner_shipping_id')
+    def _compute_fiscal_position_id(self):
+
+        def _get_fiscal_state(order):
+            """
+            Maps each order to its corresponding fiscal state based on its type,
+            fiscal conditions, and the state of the associated partner or company.
+            """
+
+            # Computing Place of Supply for particular order
+            partner_state = (
+                order.partner_id.commercial_partner_id == order.partner_shipping_id.commercial_partner_id
+                and order.partner_shipping_id.state_id
+                or order.partner_id.state_id
+            )
+            if not partner_state:
+                partner_state = order.partner_id.commercial_partner_id.state_id or order.company_id.state_id
+            if partner_state.country_id.code != 'IN':
+                partner_state = self.env.ref('l10n_in.state_in_oc', raise_if_not_found=False)
+
+            # Segrigating order based on params
+            if (
+                order.country_code != 'IN'
+                # Partner's FP takes precedence through super
+                or order.partner_shipping_id.property_account_position_id
+                or order.partner_id.property_account_position_id
+            ):
+                return False
+            elif order.l10n_in_gst_treatment == 'special_economic_zone':
+                # Special Economic Zone
+                return self.env.ref('l10n_in.state_in_oc')
```

Comment:
```suggestion
                return self.env.ref('l10n_in.state_in_oc', raise_if_not_found=False)
```
Why not using `raise_if_not_found` here?